lib/Transforms/MemCap/FoldCapIntrinsics.cpp

```diff
@@ -37,7 +38,72 @@ class MemCapFoldIntrinsics : public ModulePass {
   Function *IncOffset;
   Function *SetOffset;
   Function *GetOffset;
+  Function *GetBase;
+  Function *GetLength;
+  Function *GetType;
+  Function *GetPerms;
+  Function *GetTag;
+  Function *GetSealed;
+
   bool Modified;
+
+  template <typename Infer> void foldGet(Function *Intrinsic, Infer infer) {
```

Comment:
I think clang needs c++14 now.  If it does, replace this with a lambda with an `auto` paremeter.